### PR TITLE
Flake for the NUR

### PR DESCRIPTION
**Pull request by GuilloteauQ**
Those commits gather some "flakifications":

1. fix the flake
   this required to filter out the broken packages.

2. some packages did not have the correct platforms
   and thus failing the `nix flake check`.
   In order to determine the correct platforms, we ran
   `nix flake check --show-trace` and it tells us which
   dependency of the package cannot be built on the desired
   system.
   We then set the platforms of our package equal to the platforms
   of the limiting dependency.
   In the case that there are several dependencies with platforms
   limitations, we take the intersections between the platforms.

3. A flakes wants a set of derivations as output.
   However, the nur has some haskellPackages represented as a set.
   We cannot simply put the set in the flake output (or if we can
   I don't know how...).
   I thus created a flake for this haskellPackages and imported it
   inside the main NUR flake.
   I'm not 100percent certain this is the best choice tho...

(et un coup de `nixfmt` par bonne mesure)

### CODE REVIEW

**Comment by mpoquet** [comment]
TL;DR: Changes looks good to me but please apply the fixes below and submit another merge request :p.

- I do not think that running `nixfmt` should be done in this MR.
  This makes the useful modifications hard to see with the cosmetics changes all over the code base.
  
  BTW I'm not sure this should be run at all on `nur.nix` as it makes the code harder to read than before. Maybe a good fix would be to do `callPackage` black magic [as we did in old Kapack](https://github.com/oar-team/kapack/blob/master/default.nix#L24) so we do not need to explicitly say to use packages from nur-kapack.
- I think that the last commit (abbdbe64a8135d0b1729864c0e26d2a6f00adb78) should be split in several ones. This is better for the project maintenance and traceability if unrelated changes are kept in separate commits.
